### PR TITLE
Create dev.md

### LINKED ISSUE #5: Describe the proposed build and release process
### For Developers

1. Install VirtualBox and Vagrant on a computer with a relatively large memory size.
1. `git clone https://github.com/sql-machine-learning/desktop && cd desktop`
1. To build and start the VM: `vagrant up`
1. To SSH into the VM: `vagrant ssh`
1. The `vagrant up` command starts the VM and runs `provision.sh` as user `root`.  The `provision.sh` installs Docker and minikube and pulls related Docker images.  It also generates a `start.sh`.  Users should run `vagrant ssh` and execute `start.sh` inside the VM as user `vagrant`.  The `start.sh` starts minikube, deploys Argo/Tekton, and run all the services in the VM.
1. Every time we edit the `provison.sh`, we can run `vagrant provision` to rerun `provision.sh` as root.  This rerun would see the previously installed software.  So `provision.sh` contains some `if..else` structures to skip reinstalling software.
1. To suspend the VM, run `vagrant halt`.  You can run `vagrant up` later to resume it.
1. To completely destroy the VM and re-provision it, run `vagrant reload`.

### For Releaser

The releaser, which, in most cases, is a developer, can export a running VirtualBox VM into a VM image file with extension `.ova`.  An `ova` file is a tarball of a directory, whose content follows the OVF specification.  For the concepts, please refer to this [explanation](https://damiankarlson.com/2010/11/01/ovas-and-ovfs-what-are-they-and-whats-the-difference/).

According to this [tutorial](https://www.techrepublic.com/article/how-to-import-and-export-virtualbox-appliances-from-the-command-line/), releasers can run the following command to list running VMs.

```bash
vboxmanage list vms
```

Then, they can run the following command to export the `.ova` file.

```bash
vboxmanage export UBUNTUSERVER164 -o ubuntu_server_new.ova
```

### For End-users

To run SQLFlow on a desktop computer running Windows, Linux, or macOS, an end-user needs to download

1. the `sqlflow` command-line tool released by SQLFlow CI, and
1. the released `.ova` file.

If the end-user has VirtualBox installed -- no Vagrant required -- s/he could import the `.ova` file and start an VM.

Or, if s/he has an AWS or Google Cloud account, s/he could upload the `.ova` file to start the VM on the cloud.

Anyway, given a running VM, the end-user can run the following command to connect to it:

```bash
sqlflow -server=my-vm.aws.com:3306
```

### PR DESCRIPTION
**Pull request by wangkuiyi**
Fix https://github.com/sql-machine-learning/desktop/issues/5

### CODE REVIEW

File: dev.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+## Develop, Release, and Use SQLFlow-in-a-VM
+
+### For Developers
+
+1. Install VirtualBox and Vagrant on a computer with a relatively large memory size.
+1. `git clone https://github.com/sql-machine-learning/desktop && cd desktop`
+1. To build and start the VM: `vagrant up`
```
Comment (Yancey0623) [inline dev.md:7]: A way to speed up downloading vagrant box using tsinghua mirror 

``` bash
vagrant box add https://mirrors.tuna.tsinghua.edu.cn/ubuntu-cloud-images/bionic/current/bionic-server-cloudimg-amd64-vagrant.box --name ubuntu/bionic64
```
Ref: https://stackoverflow.com/a/22091770, https://segmentfault.com/q/1010000011063709

Comment (wangkuiyi) [reply dev.md:7]: Do you want to add instructions about using the Tsinghua mirror? I agree it is a good idea. How about we merge this and then you create a new PR to add it?

Comment (Yancey0623) [reply dev.md:7]: > How about we merge this and then you create a new PR to add it?

Sure, I'm not good at Vagrant, so I'm trying to go through as the steps from your PR and leave some comments here. I will add them to a new PR.

**Review by Yancey0623 — APPROVED** [review]
LGTM

**Pull request merged**